src/storage/mvcc/txn.rs

```diff
@@ -180,8 +180,11 @@ impl<'a> MvccTxn<'a> {
         let mut ts: u64 = u64::max_value();
         let mut versions = 0;
         let mut delete_versions = 0;
+        let mut latest_delete = None;
         while let Some((commit, write)) = try!(self.reader.seek_write(key, ts)) {
             if self.write_size >= MAX_TXN_WRITE_SIZE {
+                // Cannot remove latest delete when we haven't iterate all versions.
+                latest_delete = None;
                 break;
             }
             if !remove_older {
```

Comment:
Can you reduce the `if ... {} else {}` levels by `if ... { ... continue}`? This condition block is too long and hard to read.